### PR TITLE
Check python version that was used to install pre-commit venvs

### PR DESCRIPTION
**Pull request by potiuk**
Since we moved to Python 3.9 and started usign 3.9+ only features line functools.cache, having Python 3.8 as default python version will result in virtualenvs created with outdated python.

This pre-commit checks what is the global version of python3 and it will complain if it is lower than 3.9. This is done via explicit call of "python3", because this is what pre-commit does. Also global setting is needed because otherwise new virtualenvs created by pre-commit might be created using lower python version.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

File: scripts/ci/pre_commit/check_min_python_version.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import subprocess
+import sys
+from pathlib import Path
+
+sys.path.insert(0, str(Path(__file__).parent.resolve()))
+
+from common_precommit_utils import console
+
+required_version = (3, 9)
```
Comment (jscheffl) [inline scripts/ci/pre_commit/check_min_python_version.py]: As when we upgrade Python next time - can you make the version a bit more "catchy" such that if we search for "3.9" that we find this literal? :-D

File: scripts/ci/pre_commit/check_min_python_version.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        )
+        .stdout.strip()
+        .split("."),
+    )
+)
+
+
+if global_version < required_version:
+    console.print(f"[red]Python {required_version_str} or higher is required to install pre-commit.\n")
+    console.print(f"[green]Current version is {required_version_str}\n")
```
Comment (jscheffl) [inline scripts/ci/pre_commit/check_min_python_version.py]: Copy & Paste error?
```suggestion
    console.print(f"[green]Current version is {global_version}\n")
```

**Review by jscheffl — APPROVED** [review]
Just a bit nit :-D

File: scripts/ci/pre_commit/check_min_python_version.py (hunk 1 above)
Comment (o-nikolas) [reply scripts/ci/pre_commit/check_min_python_version.py]: +1 even a comment would do

**Review by o-nikolas — APPROVED** [review]

File: scripts/ci/pre_commit/check_min_python_version.py (hunk 1 above)
Comment (potiuk) [reply scripts/ci/pre_commit/check_min_python_version.py]: Updated it - but I am not really sure if it's better (and adding 3.9 in the comment would be quite a duplication - but i added a generic comment about the upgrade):

```python
required_version = tuple(map(int, "3.9".split(".")))
```

**Comment by potiuk** [comment]
Mypy check fixed in main.

File: scripts/ci/pre_commit/check_min_python_version.py (hunk 1 above)
Comment (jscheffl) [reply scripts/ci/pre_commit/check_min_python_version.py]: THANKS!

Comment (potiuk) [reply scripts/ci/pre_commit/check_min_python_version.py]: You are welcome!

**Pull request merged**